### DISCUSSION TITLE
How did you solve frustum culling? Can it be done with joltphysic in a way that ignores narrow phase?

### DISCUSSION THREAD

**Discussion by BreyerW** [jrouwe/JoltPhysics#538] [Q&A] [answered]
I am curious in which way you solved frustum culling in Horizon Forbidden West? Separate system on CPU or GPU? or by using JoltPhysic?

Regardless, do you think frustum culling could be done in performant way with JoltPhysic as it is right now? Frustum culling needs space partitioning that also changes over time (this part is almost always covered by physic engines and thats where idea of reusing it comes from) and a way to query that partitioning without involving narrow phase (since narrow phase is mostly for actual physic generation which is useless for frustum culling)

**Comment by jrouwe** [comment, marked as answer]
I don't think you should use a physics engine for frustum culling. Usually the physics volumes do not match the visuals / not everything visible is represented in the physics world, so you will get artifacts. I cannot comment on how frustum culling works in HFW.

**Reply by BreyerW** [reply]
Alright thanks

**Reply by BreyerW** [reply]
I know that not everything visible is in physic world but physic engines regularly have data structures for inactive physic items that dont generate physic but may or may not participate in partitioning. It is usually intened for temporarily inactive physic items  but could be used for something like frustum culling especially since frustum culling really is just special case for area trigger systems (and these tend to be included in physic engines too)
